### PR TITLE
Bug 1194295 - Partial playback broken when video does not start at zero.

### PR DESCRIPTION
**Pull request by sydvicious**
Support tests when the video does not start at zero.

### CODE REVIEW

File: media_utils/video_puppeteer.py [hunk 1]
```diff
@@ -148,6 +150,19 @@ def lag(self):
...
         return elapsed_wall_time - elapsed_current_time
 
+    def check_started(self):
+        '''
+        Note that Netflix videos don't start at zero by default, and
+        current_time does not get set until it does start.
+        '''
+        if self._start_time == 0:
+            self._start_time = self.current_time
+            self._start_time = self.current_time
```
Comment (mjzffr) [inline media_utils/video_puppeteer.py]: Line repeated twice.

File: media_utils/video_puppeteer.py [hunk 2]
```diff
@@ -148,6 +150,19 @@ def lag(self):
...
+            if self._calculated_duration:
+                self._calculated_duration = 0
```
Comment (mjzffr) [inline media_utils/video_puppeteer.py:168]: Why is `_calculated_duration` reset to 0? Could you add a comment?

File: media_utils/video_puppeteer.py [hunk 3]
```diff
@@ -148,6 +150,19 @@ def lag(self):
         elapsed_wall_time = clock() - self._start_wall_time
         return elapsed_wall_time - elapsed_current_time
 
+    def check_started(self):
+        '''
+        Note that Netflix videos don't start at zero by default, and
+        current_time does not get set until it does start.
```
Comment (mjzffr) [inline media_utils/video_puppeteer.py:156]: Confirming that I understand: `current_time` does not get set until the video starts?

File: media_utils/video_puppeteer.py [hunk 4]
```diff
@@ -148,6 +150,19 @@ def lag(self):
         elapsed_wall_time = clock() - self._start_wall_time
         return elapsed_wall_time - elapsed_current_time
 
+    def check_started(self):
```
Comment (mjzffr) [inline media_utils/video_puppeteer.py]: What do you mean by 'check_started'? Suggestion: rename to something more descriptive like `reset_playback_interval`, say.

**Comment by mjzffr** [comment]
Test results
- Passed: http://pf-jenkins.qa.mtv2.mozilla.com:8080/view/temp/job/temp_test-mn-mse-youtube-basic-y-nightly-mac/54/consoleFull
- Passed: http://pf-jenkins.qa.mtv2.mozilla.com:8080/view/temp/job/temp_test-mn-mse-youtube-default-nightly-win7_32_64/59/console
- Failed: http://pf-jenkins.qa.mtv2.mozilla.com:8080/view/temp/job/temp_test-mn-eme-netflix-nightly-win7_32_64/3/console
  - TEST-UNEXPECTED-ERROR | test_video_playback.py TestVideoPlayback.test_playback_starts | TimeoutException
  - PROCESS-CRASH | test_video_playback.py TestVideoPlayback.test_video_playback_partial | application crashed [None]

**Comment by sydvicious** [comment]
@mjzffr, feedback addressed. Please review.

**Comment by mjzffr** [comment]
@sydvicious Looks good. Please flag me for review on Bugzilla so I can r+, merge, etc.

**Pull request merged**